packages/kbn-es/src/utils/native_realm.js

```diff
@@ -36,13 +36,28 @@ exports.NativeRealm = class NativeRealm {
           `setting ${chalk.bold(username)} password to ${chalk.bold(password)}`
       );
 
-      await this._client.security.changePassword({
-        username,
-        refresh: 'wait_for',
-        body: {
-          password,
-        },
-      });
+      try {
+        await this._client.security.changePassword({
+          username,
+          refresh: 'wait_for',
+          body: {
+            password,
+          },
+        });
+      } catch (err) {
+        const isAnonymousUserPasswordChangeError =
```

Comment:
__question:__`@kbn/es` retrieves all reserved users and tries to set passwords for all of them, but this breaks when anonymous access is enabled in Elasticsearch since we can't set password for anonymous user that is also a reserved one. 

Unfortunately the name of the anonymous user is configurable so we can't easily filter out anonymous user based on the name (unless we extract the name from the settings that are being passed to `@kbn/es` and propagate it down to this method). That's why I resorted to this ugly hack with an excuse that it'll be implicitly tested by the security API integration test (e.g. that will fail if ES changes the wording of the error message).

@elastic/kibana-operations, I'm all ears if you have any better ideas!